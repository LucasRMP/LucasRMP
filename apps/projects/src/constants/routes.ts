export const routes = [
  {
    path: '/',
    label: 'Home',
  },
  {
    path: '/blog',
    label: 'Blog',
  },
  {
    path: '/projects',
    label: 'Projects',
  },
]
