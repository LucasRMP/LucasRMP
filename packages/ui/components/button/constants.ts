export type ButtonColor = 'primary' | 'success' | 'error'
export enum ButtonColorEnum {
  PRIMARY = 'primary',
  SUCCESS = 'success',
  ERROR = 'error',
}

export type ButtonVariant = 'outlined' | 'contained' | 'text'
export enum ButtonVariantEnum {
  OUTLINED = 'outlined',
  CONTAINED = 'contained',
  TEXT = 'text',
}

export type ButtonSize = 'sm' | 'md'
export enum ButtonSizeEnum {
  SMALL = 'sm',
  MEDIUM = 'md',
}

const disabledClassByVariant = {
  [ButtonVariantEnum.OUTLINED]:
    'disabled:border-gray-500 disabled:text-gray-500 disabled:hover:opacity-100',
  [ButtonVariantEnum.CONTAINED]:
    'disabled:bg-gray-500 disabled:hover:bg-gray-500',
  [ButtonVariantEnum.TEXT]: 'disabled:text-gray-500',
}

const colorToClassName = {
  [ButtonColorEnum.PRIMARY]: {
    [ButtonVariantEnum.CONTAINED]: `bg-primary hover:bg-primary-700`,
    [ButtonVariantEnum.OUTLINED]:
      'border-primary text-primary hover:text-white hover:bg-primary',
    [ButtonVariantEnum.TEXT]: 'text-primary',
  },
  [ButtonColorEnum.SUCCESS]: {
    [ButtonVariantEnum.CONTAINED]: 'bg-green-500 hover:bg-green-700',
    [ButtonVariantEnum.OUTLINED]: 'border-green-500 text-green-500',
    [ButtonVariantEnum.TEXT]: 'text-green-500',
  },
  [ButtonColorEnum.ERROR]: {
    [ButtonVariantEnum.CONTAINED]: 'bg-red-500 hover:bg-red-700',
    [ButtonVariantEnum.OUTLINED]: 'border-red-500 text-red-500',
    [ButtonVariantEnum.TEXT]: 'text-red-500',
  },
}

export const COMPOUND_VARIANTS = [
  ...Object.values(ButtonColorEnum).reduce((acc, color) => {
    Object.values(ButtonVariantEnum).forEach(variant => {
      acc.push({
        color,
        variant,
        className: `${colorToClassName[color][variant]} ${disabledClassByVariant[variant]}`,
      })
    })

    return acc
  }, [] as any[]),
]
