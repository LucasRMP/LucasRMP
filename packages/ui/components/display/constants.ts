export interface DisplayProps {
  children: React.ReactNode
  when: any
}
