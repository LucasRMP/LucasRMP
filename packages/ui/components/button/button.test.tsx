import React from 'react'
import { screen, render } from '@testing-library/react'

import { Button } from '.'
import { ButtonColor, ButtonVariant } from './constants'

function combinations<T1 = any, T2 = any>(array1: T1[], array2: T2[]) {
  const result: Array<[T1, T2]> = []
  array1.forEach(a => {
    array2.forEach(b => {
      result.push([a, b])
    })
  })
  return result
}

describe('ui/Button', () => {
  it('should render a button element with the children passed in', () => {
    render(
      <Button>
        <div data-testid="children"></div>
      </Button>,
    )

    expect(screen.getByRole('button')).toBeInTheDocument()
    expect(screen.getByTestId('children')).toBeInTheDocument()
  })

  it('should respect the fullWidth prop', () => {
    render(<Button fullWidth />)

    expect(screen.getByRole('button')).toHaveClass('w-full')
  })

  it('should apply the correct classes based on a combination of variant and color', () => {
    render(
      <>
        {combinations<ButtonColor, ButtonVariant>(
          ['primary', 'success', 'error'],
          ['contained', 'outlined', 'text'],
        ).map(([color, variant]) => (
          <Button
            key={`${color}-${variant}`}
            color={color}
            variant={variant}
            data-testid={`${color}-${variant}`}
          />
        ))}
      </>,
    )
  })
})
