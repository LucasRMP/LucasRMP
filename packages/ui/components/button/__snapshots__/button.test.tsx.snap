// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/Button should render properly in the base case 1`] = `
<button
  class="flex items-center font-semibold tracking-wider text-lh-100 rounded-md active:scale-90 transition-transform text-slate-50 py-3 px-5 text-lg bg-primary hover:bg-primary-700 disabled:bg-gray-500 disabled:hover:bg-gray-500"
  role="button"
  type="button"
>
  <div
    class="flex flex-row gap-2 text-center w-max"
  >
    A normal button
  </div>
</button>
`;

exports[`ui/Button should render properly when component is passed 1`] = `
<a
  class="flex items-center font-semibold tracking-wider text-lh-100 rounded-md active:scale-90 transition-transform text-slate-50 py-3 px-5 text-lg bg-primary hover:bg-primary-700 disabled:bg-gray-500 disabled:hover:bg-gray-500"
  href="https://google.com"
  role="button"
  type="button"
>
  <div
    class="flex flex-row gap-2 text-center w-max"
  >
    A link
  </div>
</a>
`;

exports[`ui/Button should render properly when disabled 1`] = `
<button
  class="flex items-center font-semibold tracking-wider text-lh-100 rounded-md active:scale-90 transition-transform text-slate-50 py-3 px-5 text-lg bg-primary hover:bg-primary-700 disabled:bg-gray-500 disabled:hover:bg-gray-500"
  disabled=""
  role="button"
  type="button"
>
  <div
    class="flex flex-row gap-2 text-center w-max"
  >
    A disabled button
  </div>
</button>
`;
